### PR TITLE
feature: udsink gRPC

### PR DESCRIPTION
**Pull request by chromevoid**
support udsink gRPC

Tested using the `pkg/sink/examples/udsink_log.go` example
image

### CODE REVIEW

File: pkg/sink/configs.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package sink
+
+const (
+	Protocol = "unix"
+	Addr     = "/var/run/numaflow/function.sock"
+)
```
Comment (chromevoid) [inline pkg/sink/configs.go:6]: I first refactored the configs to be pkg/configs
then when I was working on the numaflow project I found that I need to import one more pkg/configs
so I reverted the refactor and added a separate configs under sink

let me know what do you think

File: pkg/sink/types.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+// Message is used to wrap the message written to the user defined sink
+// TODO: The numaflow project use sinksdk.Message as the input argument for Apply function.
+//
+//	Therefore, since the Message type doesn't have event time and watermark, these two fields
+//	won't be populated into the Datum.event_time and Datum.watermark.
+//	In short:
+//	  Add EventTime and Watermark in the Message type
+//	  or
+//	  Delete event_time and watermark in the sink proto file.
```
Comment (chromevoid) [inline pkg/sink/types.go:11]: Please check this TODO

Comment (chromevoid) [reply pkg/sink/types.go:11]: I feel like we only need Datum...
Message type is only used in the numaflow project. We can replace [sinksdk.Message](https://github.com/numaproj/numaflow/pull/174/files#diff-9bee584308ea77899e2541fc380d19957f6bde1e5f582d960d66609516f28556R46) with sinkpb.Datum.

Comment (chromevoid) [reply pkg/sink/types.go]: removed message type..

**Comment by chromevoid** [comment]
Do we need key for sink.Datum type???

**Review by vigith — APPROVED** [review]

**Pull request merged**